frontend/src/components/pages/schemas/Schema.List.tsx

```diff
@@ -136,14 +163,26 @@ class SchemaList extends PageComponent<{}> {
 
                 {renderRequestErrors()}
 
-                <SearchBar<{ name: string }>
-                    placeholderText="Enter search term/regex"
-                    dataSource={() => (api.schemaSubjects || []).map(str => ({ name: str.name }))}
-                    isFilterMatch={this.isFilterMatch}
-                    filterText={uiSettings.schemaList.quickSearch}
-                    onQueryChanged={(filterText) => (uiSettings.schemaList.quickSearch = filterText)}
-                    onFilteredDataChanged={data => this.filteredSchemaSubjects = data}
-                />
+                <Flex alignItems="center" gap="4" mb=".5rem">
+                    <SearchField width="350px"
+                        searchText={uiSettings.schemaList.quickSearch}
+                        setSearchText={action(filterText => {
+                            uiSettings.schemaList.quickSearch = filterText;
+
+                            const searchAsNum = Number(filterText.trim());
+                            if (!isNaN(searchAsNum)) {
+                                // Keep calling it to keep the list updated
+                                // Extra calls (even when we already have data) will be automatically caught by caching
+                                this.isLoadingSchemaVersionMatches = true;
+                                api.refreshSchemaUsagesById(searchAsNum)
```

Comment:
Just a question, would it be possible to somehow fetch the schema ids with other data at the beginning for consistency instead of doing it on search?

Reply:
Requested changes mostly because of the bug above with the initial search.